source/package-manager/index.js

```diff
@@ -1,30 +1,25 @@
 import fs from 'node:fs';
 import path from 'node:path';
 import semver from 'semver';
-import {npmConfig, yarnBerryConfig, pnpmConfig, yarnConfig, npmConfigNoLockfile} from './configs.js';
+import * as configs from './configs.js';
 
 /**
 @param {string} rootDir
 @param {import('./types.d.ts').PackageManagerConfig} config
 */
-export function findLockFile(rootDir, config) {
+export function findLockfile(rootDir, config) {
 	return config.lockfiles
-		.map(filename => ({filename, filepath: path.resolve(rootDir || '.', filename)}))
-		.find(({filepath}) => fs.existsSync(filepath));
+		.map(filename => path.resolve(rootDir || '.', filename))
+		.find(filepath => fs.existsSync(filepath));
 }
 
 /**
 @param {string} rootDir
 @param {import('read-pkg').NormalizedPackageJson} pkg
 */
 export function getPackageManagerConfig(rootDir, pkg) {
-	let config = configFromPackageManagerField(pkg);
-
-	if (config === npmConfig && !findLockFile(rootDir, config)) {
-		config = npmConfigNoLockfile;
-	}
-
-	return config || configFromLockfile(rootDir) || npmConfig;
+	const config = configFromPackageManagerField(pkg);
+	return config || configFromLockfile(rootDir) || configs.npmConfig;
```

Comment:
e.g. here:

```suggestion
	return config ?? configFromLockfile(rootDir) ?? configs.npmConfig;
```

Reply:
Same answer as above. `||` increases likelihood that we'll always get a `PackageManagerConfig` from this function, and not a weird falsy value.